android/src/main/java/com/reactnativelauncharguments/LaunchArgumentsModule.java

```diff
@@ -30,20 +30,31 @@ public String getName() {
     }
 
     @Override
-    public Map<String, Object> getConstants() {
-        Map<String, Object> map = new HashMap();
+    public Map < String, Object > getConstants() {
+        Map < String, Object > map = new HashMap();
 
         Activity activity = getCurrentActivity();
         if (activity != null) {
             Intent intent = activity.getIntent();
             if (intent != null) {
-                Bundle bundle = intent.getBundleExtra("launchArgs");
-                if (bundle != null) {
-                    Set<String> ks = bundle.keySet();
-                    Iterator<String> iterator = ks.iterator();
-                    while (iterator.hasNext()) {
-                        String key = iterator.next();
-                        map.put(key, bundle.getString(key));
+                //parse all extras
+                Bundle bundleExtras = intent.getExtras();
+                if (bundleExtras != null) {
+                    for (String key: bundleExtras.keySet()) {
+                        //bundle inside bundle put by detox
+                        if ("launchArgs".equals(key)) {
+                            Bundle bundle = intent.getBundleExtra(key);
+                            if (bundle != null) {
+                                Set < String > ks = bundle.keySet();
+                                Iterator < String > iterator = ks.iterator();
+                                while (iterator.hasNext()) {
+                                    String innerkey = iterator.next();
+                                    map.put(innerkey, bundle.getString(innerkey));
+                                }
+                            }
+                        } else {
+                            map.put(key, bundleExtras.get(key));
+                        }
```

Comment:
I'm not sure if this is backward compatible with previous version, let's just leave previous version and add new logic separately?

Reply:
Yes, this is backward compatible, and needed too. This is the piece which out all key,values which are there in the intent  to the map. see https://developer.android.com/reference/android/content/Intent#getExtras() and https://stackoverflow.com/questions/6894513/how-to-get-all-extra-values-from-intent/35041128